scripts/preloadHTTPSE.js

```diff
@@ -36,81 +35,124 @@ const buildDataFiles = () => {
+    'Slashdot.org (partial)': 'redirect loop on mobile slashdot.org',
+    'Delta.com (partial)': 'https://delta.com does not redirect to https://www.delta.com',
+    'Cargo (partial)': 'breaks cargocollective.com',
+    'TMZ.com': 'breaks www.tmz.com',
+    'BusinessInsider.com (partial)': 'breaks http://www.businessinsider.com/silicon-valley-100-2016-6?op=0',
+    'Tesco (partial)': 'breaks tesco.com due to CSP mismatch',
+    'Vodafone.ie (partial)': 'breaks pagination on http://shop.vodafone.ie/shop/phonesAndPlans/phonesAndPlansHome.jsp?subPage=phones&planFilter=onAccount',
+    'iDownloadBlog (partial)': 'breaks http://www.idownloadblog.com/',
+    'Cisco.com (partial)': 'breaks http://www.cisco.com/c/m/en_us/training-events/events-webinars/techwise-tv/listings.html',
+    'GQ.com (partial)': 'mixed content on gq.com',
+    'Where 2 Get It (partial)': 'maps missing on http://us.coopertire.com/Customer-Care/Dealer-Locator.aspx?form=locator_search&addressline=92346',
+    'Thompson Hotels.com (partial)': 'missing stylesheets on http://www.thompsonhotels.com/'
   }
 
-  const rulesets = JSON.parse(fs.readFileSync('./https-everywhere/chrome/content/rulesets.json', 'utf8'))
+  const rulesets = JSON.parse(fs.readFileSync('./https-everywhere/rules/default.rulesets', 'utf8'))
 
-  // Convert XML rules to JSON
-  for (let id in rulesets.rulesetStrings) {
-    let contents = rulesets.rulesetStrings[id]
-    parseString(contents, function (err, result) {
-      if (err) {
-        throw new Error('FATAL: error parsing XML: ' + contents)
+  let jsonOutput = {
+    'rulesetStrings': [],
+    'targets': {}
+  }
+
+  for (const ruleset of rulesets) {
+    if (!ruleset.default_off && !ruleset.platform) {
+      if (ruleset.name in exclusions) {
+        console.log('NOTE: Excluding ruleset:', ruleset.name)
+        continue
       }
-      // Exclude broken rules
-      const ruleset = result.ruleset
-      if (ruleset.$.f in exclusions) {
-        console.log('NOTE: Excluding rule', JSON.stringify(result))
-        ruleset.$.default_off = exclusions[ruleset.$.f]
+
+      for (const target of ruleset.target) {
+        if (!jsonOutput['targets'][target]) {
+          jsonOutput['targets'][target] = []
+        }
+        jsonOutput['targets'][target].push(jsonOutput['rulesetStrings'].length)
       }
-      rulesets.rulesetStrings[id] = result
-    })
+
+      let r = {
+        ruleset: {
+          '$': {
+            'name': ruleset.name
+          },
+          'rule': ruleset.rule.map((rule) => {
+            return {
+              '$': {
+                'from': rule.from,
+                'to': rule.to
+              }
+            }
+          })
+        }
+      }
+
+      if (ruleset.exclusion) {
+        r.exclusion = ruleset.exclusion.map((exclusion) => {
+          return {
+            '$': {
+              'pattern': exclusion
+            }
+          }
+        })
+      }
+
+      if (ruleset.securecookie) {
```

Comment:
we don't have plans to use `securecookie` in browser-laptop so this can be omitted

Reply:
thanks for submitting a PR upstream!! I have remove the `securecookie` properties and simplified the schema in 31d9b8f.